File: core/src/main/java/com/officemanager/core/service/DesignationService.java
package com.officemanager.core.service;

import com.officemanager.core.domain.Designation;

import java.util.List;

/**
 * @author <a href="mailto:koolrich@gmail.com">Richard Nduka</a>
 */
public interface DesignationService {

    public Designation save(Designation designation);

    public Designation findById(Integer designationId);

    public List<Designation> findAll();
}

File: web/src/main/webapp/js/admin/services/DesignationService.js
/**
 * @author <a href="mailto:koolrich@gmail.com">Richard Nduka</a>
 */

'use strict'

adminModule.factory('Designation', ['$resource',
    function($resource){
        return $resource('api/designations/:id', {id: '@designationId'});
    }
]);

adminModule.factory('DesignationService', ['Designation', '$q',
    function(Designation, $q){
        return{
            findAll: function(){
                var delay = $q.defer();
                Designation.query(function(designations){
                    delay.resolve(designations);
                }, function(){
                    delay.reject('Unable to fetch designations');
                });
                return delay.promise;
            },

            findById: function(designationId){
                var delay = $q.defer();
                Designation.get({id: designationId}, function(designation){
                    delay.resolve(designation);
                }, function(){
                    delay.reject('Unable to find designation' + designationId);
                });
                return delay.promise;
            }
        }
    }
]);
